--- a/drivers/regulator/core.c
+++ b/drivers/regulator/core.c
@@ -5168,6 +5168,9 @@ regulator_register(const struct regulator_desc *regulator_desc,
 		dangling_cfg_gpiod = false;
 		dangling_of_gpiod = false;
 	}
+
+	if (rdev->desc->off_on_delay)
+		rdev->last_off_jiffy = jiffies;

 	/* register with sysfs */
 	rdev->dev.class = &regulator_class;
--- a/drivers/regulator/fixed.c
+++ b/drivers/regulator/fixed.c
@@ -159,6 +159,9 @@ static int reg_fixed_voltage_probe(struct platform_device *pdev)
 						     &drvdata->desc);
 		if (IS_ERR(config))
 			return PTR_ERR(config);
+
+		of_property_read_u32(pdev->dev.of_node,
+				     "off-on-delay-us", &drvdata->desc.off_on_delay);
 	} else {
 		config = dev_get_platdata(&pdev->dev);
 	}
--- a/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-r2s.dts
@@ -213,6 +213,7 @@ &gmac2io {
 	assigned-clock-parents = <&gmac_clk>, <&gmac_clk>;
 	clock_in_out = "input";
 	phy-handle = <&rtl8211e>;
+	phy-handle = <&yt8521s>;
 	phy-mode = "rgmii";
 	phy-supply = <&vcc_io_33>;
 	pinctrl-0 = <&rgmiim1_pins>;
@@ -236,6 +237,16 @@ rtl8211e: ethernet-phy@1 {
 			reset-assert-us = <10000>;
 			reset-deassert-us = <50000>;
 			reset-gpios = <&gpio1 RK_PC2 GPIO_ACTIVE_LOW>;
+
+		yt8521s: ethernet-phy@3 {
+			compatible = "ethernet-phy-id0000.011a",
+				     "ethernet-phy-ieee802.3-c22";
+			reg = <3>;
+			pinctrl-0 = <&eth_phy_reset_pin>;
+			pinctrl-names = "default";
+			reset-assert-us = <10000>;
+			reset-deassert-us = <50000>;
+			reset-gpios = <&gpio1 RK_PC2 GPIO_ACTIVE_LOW>;
 		};
 	};
 };
